### PR TITLE
remove unnecessary wrapping

### PR DESCRIPTION
**Pull request by berler**
**Goals (and why)**: Remove NamespacedKeyValueService and related classes such as StaticTableMappingService. These class did not provide any useful functionaltiy, especially when used with a StaticTableMappingService as this "service" just provided an identity map while doing a ton of extra work.  I have verified that these are not needed internally either.

**Implementation Description (bullets)**: Remove unnecessary code.  This will likely actually see some perf improvements as these wrappers were just wasting cpu and memory resources, and even making calls to list all table names unnecessarily.

**Concerns (what feedback would you like?)**: why weren't these removed a long time ago?

**Where should we start reviewing?**:

**Priority (whenever / two weeks / yesterday)**: normal

---
This change is [Reviewable](https://reviewable.io/reviews/palantir/atlasdb/2448)

### CODE REVIEW

**Comment by gsheasby** [comment]
See #1433 - these were suggested for removal some months ago, but back then there was some internal usage, presumably related to the large internal product. If this product no longer uses this feature, I'd be happy for it to be removed!

File: atlasdb-config/src/main/java/com/palantir/atlasdb/factory/TransactionManagers.java [hunk 1]
```diff
@@ -236,8 +235,7 @@ public static SerializableTransactionManager create(
                 atlasFactory::getTimestampService,
                 atlasFactory.getTimestampStoreInvalidator(),
                 userAgent);
-        KeyValueService kvs = NamespacedKeyValueServices.wrapWithStaticNamespaceMappingKvs(rawKvs,
-                config.initializeAsync());
+        KeyValueService kvs = rawKvs;
```
Comment (gsheasby) [inline atlasdb-config/src/main/java/com/palantir/atlasdb/factory/TransactionManagers.java]: This line is now unnecessary - simply `KeyValueService kvs = ProfilingKeyValueService.create(rawKvs, ...)` will be fine.

File: atlasdb-dagger/src/main/java/com/palantir/atlasdb/services/KeyValueServiceModule.java [hunk 2]
```diff
@@ -50,8 +49,7 @@
     public KeyValueService provideWrappedKeyValueService(@Named("rawKvs") KeyValueService rawKvs,
                                                          TimestampService tss,
                                                          ServicesConfig config) {
-        KeyValueService kvs = NamespacedKeyValueServices.wrapWithStaticNamespaceMappingKvs(rawKvs,
-                config.atlasDbConfig().initializeAsync());
+        KeyValueService kvs = rawKvs;
```
Comment (gsheasby) [inline atlasdb-dagger/src/main/java/com/palantir/atlasdb/services/KeyValueServiceModule.java]: See previous comment

File: atlasdb-tests-shared/src/test/java/com/palantir/atlasdb/schema/TableMigratorTest.java [hunk 3]
```diff
@@ -131,8 +123,7 @@ public void testMigrationToDifferentKvs() throws TableMappingNotFoundException {
         }
         checkpointer.deleteCheckpoints();
 
-        final KeyValueService verifyKvs = NamespaceMappingKeyValueService.create(
-                TableRemappingKeyValueService.create(kvs2, tableMap));
+        final KeyValueService verifyKvs = kvs2;
```
Comment (gsheasby) [inline atlasdb-tests-shared/src/test/java/com/palantir/atlasdb/schema/TableMigratorTest.java]: ditto

**Comment by berler** [comment]
I think the assertions that the class was still needed in #1433 was somewhat mistaken. Prior to #323 it was correct, but after it was not.  In between there may have been some more refactors that made it more obvious that NamespaceMappingKeyValueService actually doesn't do anything.

---







*Comments from [Reviewable](https://reviewable.io:443/reviews/palantir/atlasdb/2448#-:-Kw27gavmITyOBwcex4o:be5yrtg)*

**Comment by berler** [comment]
Review status: 0 of 11 files reviewed at latest revision, 3 unresolved discussions, some commit checks failed.

---

*[atlasdb-config/src/main/java/com/palantir/atlasdb/factory/TransactionManagers.java, line 238 at r3](https://reviewable.io:443/reviews/palantir/atlasdb/2448#-KvooYhxqlydnXoYFE7M-r3-238:-Kw27V8Ss3NQiAN9cBq8:b-896fix) ([raw file](https://github.com/palantir/atlasdb/blob/fcbe00dd0d64305dd4897f3b48c6448028c04fa5/atlasdb-config/src/main/java/com/palantir/atlasdb/factory/TransactionManagers.java#L238)):*
<details><summary><i>Previously, gsheasby (Glenn Sheasby) wrote…</i></summary><blockquote>

This line is now unnecessary - simply `KeyValueService kvs = ProfilingKeyValueService.create(rawKvs, ...)` will be fine.
</blockquote></details>

Done.

---

*[atlasdb-dagger/src/main/java/com/palantir/atlasdb/services/KeyValueServiceModule.java, line 52 at r3](https://reviewable.io:443/reviews/palantir/atlasdb/2448#-KvooYhxqlydnXoYFE7N-r3-52:-Kw27X1ufNfjI6A-p4Om:b-896fix) ([raw file](https://github.com/palantir/atlasdb/blob/fcbe00dd0d64305dd4897f3b48c6448028c04fa5/atlasdb-dagger/src/main/java/com/palantir/atlasdb/services/KeyValueServiceModule.java#L52)):*
<details><summary><i>Previously, gsheasby (Glenn Sheasby) wrote…</i></summary><blockquote>

See previous comment
</blockquote></details>

Done.

---

*[atlasdb-tests-shared/src/test/java/com/palantir/atlasdb/schema/TableMigratorTest.java, line 126 at r3](https://reviewable.io:443/reviews/palantir/atlasdb/2448#-KvooYhxqlydnXoYFE7T-r3-126:-Kw27Y-GxqqnrPCGktus:b-896fix) ([raw file](https://github.com/palantir/atlasdb/blob/fcbe00dd0d64305dd4897f3b48c6448028c04fa5/atlasdb-tests-shared/src/test/java/com/palantir/atlasdb/schema/TableMigratorTest.java#L126)):*
<details><summary><i>Previously, gsheasby (Glenn Sheasby) wrote…</i></summary><blockquote>

ditto
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/palantir/atlasdb/2448)*

**Comment by gsheasby** [comment]
:lgtm:

---

Reviewed 6 of 13 files at r1, 1 of 1 files at r2, 1 of 3 files at r3, 3 of 3 files at r4.
Review status: all files reviewed at latest revision, all discussions resolved, some commit checks failed.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/palantir/atlasdb/2448#-:-Kw4AyIwvqiWmfj3OHqw:bnfp4nl)*

File: atlasdb-config/src/main/java/com/palantir/atlasdb/factory/TransactionManagers.java [hunk 4]
```diff
@@ -236,9 +235,7 @@ public static SerializableTransactionManager create(
                 atlasFactory::getTimestampService,
                 atlasFactory.getTimestampStoreInvalidator(),
                 userAgent);
-        KeyValueService kvs = NamespacedKeyValueServices.wrapWithStaticNamespaceMappingKvs(rawKvs,
-                config.initializeAsync());
-        kvs = ProfilingKeyValueService.create(kvs, config.getKvsSlowLogThresholdMillis());
+        KeyValueService kvs = ProfilingKeyValueService.create(rawKvs, config.getKvsSlowLogThresholdMillis());
```
Comment (jeremyk-91) [inline atlasdb-config/src/main/java/com/palantir/atlasdb/factory/TransactionManagers.java]: Confirmed this is ok - the namespaced key value service's initialization thankfully doesn't seem to have side effects on the kvs that might be needed in later stages.

**Pull request merged**